### DISCUSSION TITLE
OSCLinkProvider

### DISCUSSION THREAD

**Discussion by openmax** [sourcelair/xterm.js#5068] [Q&A]
This is the implementation in xterm.js of the OSCLinkProvider :
https://github.com/xtermjs/xterm.js/blob/master/src/browser/OscLinkProvider.ts

if there isn't a linkHandler it uses a defaultActivate where a STRONG WARNING push up appares, that advice you about the risk
if you open the url ...

`activate: (e, text) => (linkHandler ? linkHandler.activate(e, text, range) : defaultActivate(e, text))`

There is an example on how implement a linkHandler to prevent this on a selected list of url ? 

Example :
in my terminal :
>echo -e `click here: \x1b]8;;https://xtermjs.org\x07OSC 8 hyperlink text\x1b]8;;\x07`

than i have a link to https://xtermjs.org/ that without the use of linkHandler it push out :

        Do you want to navigate to ${uri}?\n\nWARNING: This link could potentially be dangerous

**Comment by jerch** [comment]
See docs: https://github.com/xtermjs/xterm.js/blob/f77a763858e8dcba84f5595b5342554336d82853/typings/xterm.d.ts#L157-L168

**Comment by openmax** [comment]
yes, sorry but my typescript/javascript level is zero , 
so i'd like to see an example of how a scheleton "IlinkHandler" function can be done ...
